File: packages/Server/Domain/Models/Identity.cs
﻿using System;
using System.ComponentModel.DataAnnotations;

namespace Solidarity.Domain.Models
{
	public struct IdentityRoles
	{
		public const string Admin = "Admin";
		public const string Member = "Member";
	}

	public class Identity : Model
	{
		public int AccountId { get; set; }

		public Account Account { get; set; } = null!;

		[MaxLength(50)]
		public string FirstName { get; set; } = null!;

		[MaxLength(50)]
		public string LastName { get; set; } = null!;

		public DateTime? BirthDate { get; set; }
	}
}

File: packages/Client/sdk/models/Identity.ts
import { Account, Model, model } from "sdk"

export const enum IdentityRoles {
	Admin = 'admin',
	Member = 'member',
}

@model('Identity')
export class Identity extends Model {
	accountId?: number
	account?: Account
	firstName?: string
	lastName?: string
	birthDate?: Date
}